mteb/tasks/Retrieval/multilingual/MLQARetrieval.py

```diff
@@ -0,0 +1,162 @@
+    "mlqa.vi.es": ["vie-Latn", "spa-Latn"],
+    "mlqa.vi.hi": ["vie-Latn", "hin-Deva"],
+    "mlqa.vi.vi": ["vie-Latn", "vie-Latn"],
+    "mlqa.vi.zh": ["vie-Latn", "cmn-Hans"],
+    "mlqa.zh.ar": ["cmn-Hans", "ara-Arab"],
+    "mlqa.zh.de": ["cmn-Hans", "deu-Latn"],
+    "mlqa.zh.en": ["cmn-Hans", "eng-Latn"],
+    "mlqa.zh.es": ["cmn-Hans", "spa-Latn"],
+    "mlqa.zh.hi": ["cmn-Hans", "hin-Deva"],
+    "mlqa.zh.vi": ["cmn-Hans", "vie-Latn"],
+    "mlqa.zh.zh": ["cmn-Hans", "cmn-Hans"],
+}
+
+_ISO6393_to_ISO6391 = {
+    "ara-Arab": "ar",
+    "deu-Latn": "de",
+    "eng-Latn": "en",
+    "spa-Latn": "es",
+    "hin-Deva": "hi",
+    "vie-Latn": "vi",
+    "zho-Hans": "zh",
+}
+
+
+def extend_lang_pairs() -> dict[str, list[str]]:
+    # add all possible language pairs
+    hf_lang_subset2isolang = {}
+    for x in _ISO6393_to_ISO6391.keys():
+        for y in _ISO6393_to_ISO6391.keys():
+            if x != y:
+                pair = f"{x}_{y}"
+                hf_lang_subset2isolang[pair] = [
+                    x,
+                    y,
+                ]
+    return hf_lang_subset2isolang
+
+
+_EVAL_LANGS = extend_lang_pairs()
+
+
+class MLQARetrieval(AbsTaskRetrieval, CrosslingualTask):
+    metadata = TaskMetadata(
+        name="MLQARetrieval",
+        description="""MLQA (MultiLingual Question Answering) is a benchmark dataset for evaluating cross-lingual question answering performance.
+        MLQA consists of over 5K extractive QA instances (12K in English) in SQuAD format in seven languages - English, Arabic,
+        German, Spanish, Hindi, Vietnamese and Simplified Chinese. MLQA is highly parallel, with QA instances parallel between
+        4 different languages on average.""",
+        reference="https://huggingface.co/datasets/mlqa",
+        dataset={
+            "path": "mlqa",
+            "revision": "397ed406c1a7902140303e7faf60fff35b58d285",
+        },
+        type="Retrieval",
+        category="s2p",
+        eval_splits=["validation", "test"],
+        eval_langs=_EVAL_LANGS,
+        main_score="ndcg_at_10",
+        date=None,
+        form=None,
+        domains=None,
+        task_subtypes=None,
+        license=None,
+        socioeconomic_status=None,
+        annotations_creators=None,
+        dialect=None,
+        text_creation=None,
+        bibtex_citation="""@article{lewis2019mlqa,
+        title = {MLQA: Evaluating Cross-lingual Extractive Question Answering},
+        author = {Lewis, Patrick and Oguz, Barlas and Rinott, Ruty and Riedel, Sebastian and Schwenk, Holger},
+        journal = {arXiv preprint arXiv:1910.07475},
+        year = 2019,
+        eid = {arXiv: 1910.07475}
+        }""",
+        n_samples=None,
+        avg_character_length=None,
+    )
+
+    def load_data(self, **kwargs):
+        if self.data_loaded:
+            return
+
+        _dataset_raw = {}
+        self.queries, self.corpus, self.relevant_docs = {}, {}, {}
+
+        for lang in _LANGUAGES.keys():
```

Comment:
This loop loads all language pairs, I'll update it to use `self.langs` and upload only specified language pairs